server/glsp-server/src/main/java/com/eclipsesource/glsp/server/actionhandler/DIActionDispatcher.java

```diff
@@ -19,20 +19,39 @@
 
 import com.eclipsesource.glsp.api.action.Action;
 import com.eclipsesource.glsp.api.action.ActionDispatcher;
+import com.eclipsesource.glsp.api.action.ActionMessage;
 import com.eclipsesource.glsp.api.handler.ActionHandler;
+import com.eclipsesource.glsp.api.jsonrpc.GLSPClient;
+import com.eclipsesource.glsp.api.jsonrpc.GLSPClientProvider;
 import com.eclipsesource.glsp.api.provider.ActionHandlerProvider;
 import com.google.inject.Inject;
 
 public class DIActionDispatcher implements ActionDispatcher {
 
+	@Inject
+	protected GLSPClientProvider clientProvider;
+	
 	@Inject
 	protected ActionHandlerProvider handlerProvider;
-
+	
+	@Override
 	public Optional<Action> dispatch(String clientId, Action action) {
 		Optional<ActionHandler> handler = handlerProvider.getHandler(action);
 		if (handler.isPresent()) {
 			return handler.get().execute(clientId, action);
 		}
 		return Optional.empty();
 	}
+	
+	@Override
+	public void send(String clientId, Action action) {
+		GLSPClient client = clientProvider.resolve(clientId);
+		if (client == null) {
+			System.err.println("Client not initialized yet; unable to trigger server-to-client notification");
```

Comment:
Please use log4j (Logger.error()) for this

Reply:
I used the Syserr because I wasn't sure if we were really expecting to run into this case or if this was really and error; but after thinking about it I now believe it should be an IllegalStateException. The entry point, after connecting to a new client, should always be the LoadModelAction, which would initialize the clientId. And the server has no reason to send anything to a client for a model that isn't loaded

So I'll revert to the IllegalStateException